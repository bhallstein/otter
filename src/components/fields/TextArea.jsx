import React from 'react';
import Context__PageData from '../Context__PageData';


export default class TextArea extends React.Component {

  constructor(props) {
    super(props);

    this.textarea_ref = React.createRef();
    this.props.data_channel.handler = this.get_data.bind(this);
  }


  get_data() {
    return this.textarea_ref.current.value;
  }


  render() {
    return (
      <Context__PageData.Consumer>{(ctx) => (
        <textarea ref={this.textarea_ref} value={this.state.value} onChange={ctx.should_update} />
      )}</Context__PageData.Consumer>
    );
  }

}