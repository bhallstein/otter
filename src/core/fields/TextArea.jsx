import React from 'react';
import PageDataContext from '../PageDataContext';
import FieldLabel from '../other/FieldLabel';
import Utils from '../definitions/utils';


export default class TextArea extends React.Component {

  constructor(props) {
    super(props);
    this.cb_change = this.cb_change.bind(this);
  }

  cb_change(ev) {
    this.props.containing_data_item[this.props.field_def.name] = ev.target.value;
    this.setState({});
    this.ctx.value_updated();
  }


  render() {
    const field_def            = this.props.field_def;
    const containing_data_item = this.props.containing_data_item;
    const is_top_level         = this.props.is_top_level;
    const ContextConsumer      = this.props.consumer_component || PageDataContext.Consumer;
    const value                = containing_data_item[field_def.name];
    const mono                 = field_def.mono || false;
    const label                = field_def.description || Utils.humanify_str(field_def.name);

    return (
      <ContextConsumer>{ctx => (this.ctx = ctx) && (
        <div className="field">

          <FieldLabel label={label} is_top_level={is_top_level} />

          <textarea className={`textarea ${mono ? 'otter-monospace': ''}`}
                    style={mono ? { fontSize: '0.9em' } : {}}
                    ref={this.textarea_ref} value={value || ''} onChange={this.cb_change} />

        </div>
      )}</ContextConsumer>
    );
  }

}

