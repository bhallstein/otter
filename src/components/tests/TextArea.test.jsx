import React from 'react';
import TextArea from '../generic/TextArea';
import TestRenderer from 'react-test-renderer';

it('renders correctly', () => {
  // mocks
  const values = {
    content: '',
  };
  const onChange = () => {
    return;
  }

  const tree = TestRenderer
    .create(<TextArea action={onChange} values={values}></TextArea>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});