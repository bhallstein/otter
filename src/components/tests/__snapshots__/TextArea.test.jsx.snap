// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<textarea
  className="c-textarea"
  onBlur={[Function]}
  onChange={[Function]}
/>
`;
